test/integration/test_integration__io.py

```diff
@@ -1107,3 +1107,18 @@ def test_non_rectilinear__skip_parse_coordinates(open_rasterio):
     assert xds.rio.shape == (10, 10)
     with rasterio.open(test_file) as rds:
         assert rds.transform == xds.rio.transform()
+
+
+def test_complex_dtype(tmp_path):
+    test_file = os.path.join(TEST_INPUT_DATA_DIR, "cint16.tif")
+    xds = rioxarray.open_rasterio(test_file)
+    assert xds.rio.shape == (100, 100)
+    assert xds.dtype == "complex64"
+
+    # NOTE: waiting on https://github.com/mapbox/rasterio/issues/2206
+    # tmp_output = tmp_path / "tmp_cint16.tif"
+    # with pytest.warns(SerializationWarning):
+    #    xds.rio.to_raster(str(tmp_output), dtype='complex_int16')
+    #    with rasterio.open(str(tmp_output)) as riofh:
+    #        print(riofh.profile)
+    #        assert riofh.dtype == 'complex_int16'
```

Comment:
Wanted to test the round trip to ensure CInt16 loaded with rioxarray is also written to a CInt16 tif and came across an issue with nodata handling https://github.com/mapbox/rasterio/issues/2206